mifosng-android/src/main/java/com/mifos/utils/CheckSelfPermissionAndRequest.java

```diff
@@ -128,8 +131,16 @@ public void onClick(DialogInterface dialog, int which) {
                                                 R.string.package_name), activity.getPackageName()
                                                 , null);
                                         intent.setData(uri);
-                                        activity.startActivityForResult(intent,
-                                                Constants.REQUEST_PERMISSION_SETTING);
+                                        try {
```

Comment:
A safe rule of thumb while starting any activity is to first check using `PackageManager` and `Intent.resolveActivity()` whether some activity exists to handle that intent. I think instead of try-catch, you should have used `if (activity exists) {start activity} else {show message}`. What do you say @therajanmaurya ?

It's quite rare though that Settings activity is not found or modded somehow in a way that the usual approach would lead to crash!